spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubsubAutoConfiguration.java

```diff
@@ -48,49 +51,55 @@
 	public static final int DEFAULT_EXECUTOR_THREADS = 4;
 
 	@Bean
-	@ConditionalOnMissingBean
-	public ExecutorProvider executorProviderForPublisher() {
+	public ExecutorProvider publisherExecutorProvider() {
```

Comment:
Sorry, I misled you a bit with method reference dependency inject.
So, right, the use may provide his/her own `ExecutorProvider` on the matter.
So, we should keep `@ConditionalOnMissingBean` and use argument injection as you had before.